integration/utils/DriverFactory.js

```diff
@@ -0,0 +1,56 @@
+const { Builder } = require('selenium-webdriver');
+const config = require('./config');
+
+class DriverFactory {
+  configure(host) {
+    let builder = new Builder();
+    const sauceLabsURL = 'https://ondemand.saucelabs.com/wd/hub';
+
+    switch (host) {
+      case 'sauce-firefox':
+        builder.usingServer(sauceLabsURL);
+        builder.withCapabilities({
+          ...config.firefoxOptions,
+          ...config.sauceOptions
+        });
+        break;
+      case 'sauce-chrome':
+        builder.usingServer(sauceLabsURL);
+        builder.withCapabilities({
+          ...config.chromeOptions,
+          ...config.sauceOptions
+        });
+        break;
+      case 'firefox':
+        builder.forBrowser('firefox');
+        builder.withCapabilities(config.firefoxOptions);
+        break;
+      case 'chrome':
+        builder.forBrowser('chrome');
+        builder.withCapabilities(config.chromeOptions);
+        break;
+    }
+    return builder;
+  }
+
+  async build(testName, host = this.config.host) {
+    this.host = host;
+    this.driver = await this.configure(host).build();
+    const { id_ } = await this.driver.getSession();
+    this.sessionId = id_;
+    this.driver.executeScript('sauce:job-name=' + testName);
```

Comment:
Do we have to check for `    if (this.host.startsWith('sauce')) {` here?

Reply:
Yes, so the test can run either locally or on Sauce Labs.
The video URL only exists when the test running on Sauce Labs.

In `configure(host)` function, if host starts with 'sauce', e.g. 'sauce-chrome'. The test will run on Sauce Labs using latest Chrome. Other wise if hose is `chrome` it will run locally.